src/PhpImap/Mailbox.php

```diff
@@ -326,7 +327,20 @@ public function setAttachmentsIgnore(bool $attachmentsIgnore): void
     {
         $this->attachmentsIgnore = $attachmentsIgnore;
     }
-
+	
+    /**
+     * Set $this->setAttachmentsRandomFilename param. 
+     *
+     * @param int $random ATTACH_FILE_NAMERANDOM, ATTACH_FILE_NAMEORIGINAL
+     *
+     * @return Mailbox
+     */
+    public function setAttachmentsFilenameMode(int $mode) : Mailbox
+    {
+        $this->attachmentsFilenameMode = $mode;
```

Comment:
I miss a check here, which ensures, that only supported modes can be set. Something like this:
```suggestion
        if (!defined($mode))
        {
                throw new UnexpectedValueException("The defined mode '$mode' is not supported. Supported modes: ATTACH_FILE_NAMERANDOM, ATTACH_FILE_NAMEORIGINAL");
        }

        $this->attachmentsFilenameMode = $mode;
```

Note: I haven't tested this code.